### PR TITLE
Quantile regression with BART

### PR DESCRIPTION
**Pull request by aloctavodia**
+ [x] Notebook follows style guide https://docs.pymc.io/en/latest/contributing/jupyter_style.html
+ [x] Check the notebook is not excluded from any pre-commit check: https://github.com/pymc-devs/pymc-examples/blob/main/.pre-commit-config.yaml

### CODE REVIEW

File: examples/case_studies/BART_quantile_regression.ipynb [hunk 1]
```diff
@@ -0,0 +1,666 @@
+{
```
Comment (juanitorduz) [inline examples/case_studies/BART_quantile_regression.ipynb:1]: <p>Shall we add <code>%config InlineBackend.figure_format = "retina"</code> to have higher quality on the images? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/510/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by juanitorduz** [comment]
Besides the comment above the notebook looks great! Nice addition! I did not about this application of the asymmetric Laplace distribution.

**Comment by aloctavodia** [comment]
Thanks for the comment I updated the notebook addressing it and also change a little bit the model, so we only need to fit one model. I found this use for the asymmetric Laplace very recently, too! I am still surprised, haha.

**Review by juanitorduz — APPROVED** [review]
The model "compression" makes easier to read :)

File: examples/case_studies/BART_quantile_regression.myst.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+from scipy import stats
+
+print(f"Running on PyMC v{pm.__version__}")
+```
+
+```{code-cell} ipython3
+%config InlineBackend.figure_format = "retina"
+RANDOM_SEED = 5781
+np.random.seed(RANDOM_SEED)
```
Comment (OriolAbril) [inline examples/case_studies/BART_quantile_regression.myst.md:39]: Is this for the bart submodule/library not using generators/random states yet?

File: examples/case_studies/BART_quantile_regression.myst.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+plt.legend();
+```
+
+We are going to use a simple dataset to model the Body Mass Index for Dutch kids and young men as a function of their age.
+
+```{code-cell} ipython3
+try:
+    bmi = pd.read_csv(Path("..", "data", "bmi.csv"))
+except FileNotFoundError:
+    bmi = np.loadtxt(pm.get_data("bmi.csv"))
```
Comment (OriolAbril) [inline examples/case_studies/BART_quantile_regression.myst.md:70]: I think this won't work, np.loadtxt will return and array and the next line `bmi.plot` will fail.

File: examples/case_studies/BART_quantile_regression.myst.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+quiroga2022bart
+:::
+
++++
+
+## Watermark
+
+```{code-cell} ipython3
+%load_ext watermark
+%watermark -n -u -v -iv -w
```
Comment (OriolAbril) [inline examples/case_studies/BART_quantile_regression.myst.md:167]: can you add the `-p pytensor,xarray` flag?

**Review by OriolAbril — APPROVED** [review]
left three minor comments, could be merged as is.

As always, note I am reviewing on the myst file but you should update the notebook, then have pre-commit update the myst file

File: examples/case_studies/BART_quantile_regression.myst.md (hunk 2 above)
Comment (aloctavodia) [reply examples/case_studies/BART_quantile_regression.myst.md:39]: Yes

**Pull request merged**